components/VercelChat/messages.tsx

```diff
@@ -35,17 +32,25 @@ export function Messages({
   children,
 }: MessagesProps) {
   const messagesRef = useRef<HTMLDivElement>(null);
-  const messagesLength = useMemo(() => messages.length, [messages]);
+
+  const { ref, inView } = useInView({
+    root: messagesRef.current,
+    threshold: 0,
+  });
 
   useEffect(() => {
-    if (messagesRef.current) {
-      messagesRef.current.scrollTop = messagesRef.current.scrollHeight;
+    const scrollContainer = messagesRef.current;
+    if (scrollContainer && inView) {
+      // Use requestAnimationFrame to scroll after the next browser paint
+      requestAnimationFrame(() => {
+        scrollContainer.scrollTop = scrollContainer.scrollHeight;
+      });
     }
-  }, [messagesLength]);
+  }, [messages, status, inView]); // Add status to dependencies for streaming updates
```

Comment:
if your PR is for `mermaid chart`, why are you modifying `scroll behavior` of messages?